### PR TITLE
[Salesforce] Bulk Update

### PR DESCRIPTION
**Pull request by nick-Ag**
This PR adds support for the Salesforce Bulk API Update operation. This is an initial build that will only be useable by Segment Internal users. More details on that can be found in the [(Segment Internal) Bulk API Mini-SDD](https://paper.dropbox.com/doc/Mini-SDD-Salesforce-Bulk-API--BhoQssc8VEjaaGVvnkxwnG0EAg-OW8Dn2fzQRDPlCa8trLLG).

## Testing
Testing completed successfully via stage deploy. Batches of events arrive in the `bulkUpdate` handler and are correctly delivered downstream to Salesforce.

Tested with the following Python script:
```python
import analytics
import csv
import time

analytics.write_key = '<write key>'
analytics.host = 'https://api.segment.build'

with open('./lead_id.csv') as file:
    reader = csv.reader(file)
    i = 0
    for row in reader:
        lead_id = row[0]
        print(i, lead_id)
        time.sleep(.05)
        analytics.track(str(i), 'Lead', {
            'recordId': lead_id,
            'company': 'Bulk Update Test Stage 6.9.' + str(i),
            'last_name': 'Stage 6.9.' + str(i),
            'email': 'stage@test.' + str(i)
        })

        i += 1

    print('flushing...')
    analytics.flush()
    print('done')
``` 
`lead_id.csv` is a list of all Lead object Ids in my SF workspace.

Batch handler triggering in DataDog:
Screen Shot 2022-06-09 at 10 21 34 AM

Salesforce Bulk Jobs being triggered and processed:
Screen Shot 2022-06-09 at 10 21 11 AM

Event Delivery in stage:
Screen Shot 2022-06-09 at 10 20 55 AM

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/salesforce/account/index.ts [hunk 1]
```diff
@@ -198,9 +207,10 @@ const action: ActionDefinition<Settings, Payload> = {
       }
 
       return await sf.bulkUpsert(payload, OBJECT_NAME)
+    } else if (payload[0].operation === 'bulkUpdate') {
```
Comment (brennan) [inline packages/destination-actions/src/destinations/salesforce/account/index.ts:210]: The `performBatch` handlers are very similar across all Salesforce actions - I wonder if there's a way to abstract this into a single handler so all the batch logic lives in one place?

Comment (nick-Ag) [reply packages/destination-actions/src/destinations/salesforce/account/index.ts:210]: Something like an `sf.handleBatch` method that would replace the if statements in each block? I'll explore that it sounds cleaner

File: packages/destination-actions/src/destinations/salesforce/sf-utils.ts [hunk 2]
```diff
@@ -110,23 +116,36 @@ const buildCSVFromHeaderMap = (
...
+
+    const uniqueIdValue = getUniqueIdValue(payloads[i])
+    rows += `${row}"${uniqueIdValue}"\n`
   }
   return rows
 }
 
+const getUniqueIdValue = (payload: GenericPayload): string => {
+  if (payload.operation === 'bulkUpsert') {
+    return payload.bulkUpsertExternalId?.externalIdValue as string
```
Comment (rhall-twilio) [inline packages/destination-actions/src/destinations/salesforce/sf-utils.ts:128]: If bulkUpsertExternalId is undefined, should this return NO_VALUE?

**Review by rhall-twilio — APPROVED** [review]

File: packages/destination-actions/src/destinations/salesforce/sf-utils.ts (hunk 2 above)
Comment (nick-Ag) [reply packages/destination-actions/src/destinations/salesforce/sf-utils.ts]: Very true, I've replace that `return NO_VALUE` with throwing an error instead as those fields are required

**Pull request merged**